services/navigation/builtin/builtin_test.go

```diff
@@ -385,6 +387,19 @@ func TestStartWaypoint(t *testing.T) {
 				wp2, err = ns.(*builtIn).store.NextWaypoint(ctx)
 				test.That(t, err, test.ShouldBeNil)
 			}
+
+			// ensure we actually start the wp2 waypoint before removing it
+			ready := false
+			for !ready {
+				ns.(*builtIn).mu.RLock()
+				svcWp := ns.(*builtIn).waypointInProgress
+				ns.(*builtIn).mu.RUnlock()
+				if wp2.ID != svcWp.ID {
+					time.Sleep(time.Millisecond)
+				} else {
+					ready = true
+				}
+			}
```

Comment:
Use testutils.WaitForAssertion() (from goutils/testutils) if possible here. (Or WaitForAssertionWithSleep() if you need tighter loops than 50ms.)

Reply:
This is neat. I didn't know that existed.